### PR TITLE
fix dict and tsv merging

### PR DESCRIPTION
**Pull request by JuliaSprenger**
* add dedicated merge_dfs_by_index function + tests
* improve dictionary merging
* add tests

### CODE REVIEW

File: ando/tools/generator/tests/test_utils.py [hunk 1]
```diff
@@ -20,44 +23,45 @@ def test_create_file(self):
...
-            "d": [2]
+            "i": [1, 2, 3, 4],
+            "a": ['a1', 'a2', 'a3', None],
+            "b": [None, 'b2', None, 'b4']
         })
+        a.set_index('i', inplace=True)
         b = pd.DataFrame({
-            "a": [0],
-            "b": [1],
-
```
Comment (Slowblitz) [inline ando/tools/generator/tests/test_utils.py:41]: Why those changes ?

File: ando/tools/generator/utils.py [hunk 2]
```diff
@@ -52,10 +54,11 @@ def merge_dict(original_data, new_data):
     """
     Merge two dictionaries.
    
-   Merge the content of a `new` dictionary into another, `original` dictionary. A new 
-   dictionary with the merged content is created. Values are preserved and a ValueError is
-   raised if incompatible content is encountered. Overlapping lists are extended and nested
-   dictionaries are merged recursively.
+    Merge the content of a `new` dictionary into another, `original` dictionary.
```
Comment (Slowblitz) [inline ando/tools/generator/utils.py:57]: AH ! yes good eyes =)

File: ando/tools/generator/utils.py [hunk 3]
```diff
@@ -67,30 +70,69 @@ def merge_dict(original_data, new_data):
...
+
+    # check for contradicting values by comparing A+B with B+A
+    left_combine = original_df.combine_first(new_df)
+    right_combine = new_df.combine_first(original_df)
+
+    if not left_combine.equals(right_combine):
+        raise ValueError('Dataframes have incompatible values: '
+                         f'{left_combine.compare(right_combine)}')
+
+    return right_combine
```
Comment (Slowblitz) [inline ando/tools/generator/utils.py:137]: so we re not using the one already provide in pandas ?

File: ando/tools/generator/tests/test_utils.py (hunk 1 above)
Comment (JuliaSprenger) [reply ando/tools/generator/tests/test_utils.py:41]: I made the example a bit more realistic in the sense that there's more than 2 columns. With the new examples we can cover all these cases:
- none of the two dfs have a value for a field
- only one of the two dfs has a value for a field

Also I now ensure that column 0 is always used as index and therefore is also the basis for merging multiple dataframes (see `merge_dfs_by_index` below). The core mechanics of the merge are tested for that function explicitely, that's why I had to add a lot of test functions. Here it's mostly about writing to the file.

File: ando/tools/generator/utils.py (hunk 3 above)
Comment (JuliaSprenger) [reply ando/tools/generator/utils.py:137]: yes, we do as basically this is only thin wrapper for the df.combine_first() method

**Comment by JuliaSprenger** [comment]
@Slowblitz Sorry, I accidentally pushed directly to your branch instead of mine, so this PR is now considered merged because the commits are already included in your branch. However I think we can continue the discussion here and I will address any issues by pushing directly to your branch. Ok?

**Pull request merged**